core/arch/arm/kernel/secure_partition.c

```diff
@@ -123,21 +106,25 @@ struct sp_session *sp_get_session(uint32_t session_id)
 	return NULL;
 }
 
-TEE_Result sp_partition_info_get_all(struct ffa_partition_info *fpi,
-				     size_t *elem_count)
+TEE_Result sp_partition_info_get(struct ffa_partition_info *fpi,
+				 const TEE_UUID *ffa_uuid, size_t *elem_count)
 {
 	size_t in_count = *elem_count;
 	struct sp_session *s = NULL;
 	size_t count = 0;
 
 	TAILQ_FOREACH(s, &open_sp_sessions, link) {
-		if (s->state == sp_dead)
-			continue;
-		if (count < in_count) {
-			spmc_fill_partition_entry(fpi, s->endpoint_id, 1);
-			fpi++;
+		if (!ffa_uuid || !memcmp(&s->ffa_uuid, ffa_uuid,
```

Comment:
You could do:
``` C
if (ffa_uuid &&
    memcmp(&s->ffa_uuid, ffa_uuid, sizeof(*ffa_uuid))
         continue;
```
instead and leave the lines below unchanged, that should make the code a bit easier to read.

Reply:
Ack